draft 2.2 release notes (#3745)
influxdata/docs-v2: content/influxdb/v2.2/reference/release-notes/influxdb.md

weight: 101
---

> ## v2.2 [2022-02-17]

Reviewer: I don't see any of these, from the changelog:

1. [a40e12b](https://github.com/influxdata/influxdb/commit/a40e12b): Allow changing a password with `influxd recovery user update`
1. [b02c89e](https://github.com/influxdata/influxdb/commit/b02c89e): Option to log flux queries cancelled because of server shutdown
1. [3ec5a57](https://github.com/influxdata/influxdb/commit/3ec5a57): Tell browser about cookie expiry
1. [df01d93](https://github.com/influxdata/influxdb/commit/df01d93): Allow flux http calls to be unlimited
1. [49ce57c](https://github.com/influxdata/influxdb/commit/49ce57c): Remove telegraf endpoint pagination

Author: 1. Added "Recover user credentials" on line 93.  Will also add docs issue, and send docs PR review separately.
2. Looks like line 36 - do we say somewhere how to view this log?
3. Is line 136 OK?
4. Is line137 OK?
5. Is line 138 OK?

cc @samhld

Author: @lesam, please let us know if you have any further edits to this PR for Monday. @samhld has approved, so if nothing further, will merge this PR into our base branch and call it good.

Author: thank you!

This release includes the following new [features](#features) and several [bug fixes](#bug-fixes).

### Features

- [Technical preview of remote data replication](#replicate-data-remotely-technical-preview)
- [Flux updates](#flux-updates)

> - [Build maintenance](#build-maintenance)

Reviewer 2: this was added- see https://github.com/influxdata/docs-v2/issues/3866

- [Task metadata](#task-metadata)
- [Metrics](#metrics)
- [Display runtime configuration in use by `influxd`](#display-runtime-configuration-in-use-by-influxd)

#### Technical preview: Durable subscription to replicate data

- Add the option to [Replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).

  This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature allows you to create a durable subscription on a per bucket basis for the purposes of replicating data from an InfluxDB OSS instance to InfluxDB Cloud. 

>   -  Allow users to store, analyze and aggregate data locally while also forwarding newly arriving data to a centralized InfluxDB Cloud account. 

Reviewer 3: ```suggestion
  -  Allow users to store, analyze, and aggregate data locally while also forwarding newly arriving data to a centralized InfluxDB Cloud account. 
```

   - `PointsFailedToQueue`
   - `BytesFailedToQueue`

- Add the following metrics for writes, to indicate the status:
  - `pointsWriteRequested`
	- `pointsWriteOk`
	- `pointsWriteDropped`
	- `pointsWriteErr`
	- `timeout`

- Add disk size metrics per shard:
  - `total`: number of files per shard
  - `disk_bytes`: data size in bytes for each shard

- Add cache subsystem metrics:
  - `inuse_bytes`: Current memory consumption of cache
  - `diskBytes`: Size of most recent snapshot"
  - `latest_snapshot`: Unix time of most recent snapshot
  - `writes_total`: Counter of all writes to cache
  - `writes_err`: Counter of failed writes to cache
  - `writes_dropped`: Counter of writes (with any dropped points) to cache

- Add WAL subsystem metrics:
  - `size`: Size of WAL in bytes
  - `writes`: Number of write attempts to the WAL
  - `writes_err`: Number of failed write attempts to the WAL

See [InfluxDB OSS metrics](/influxdb/v2.1/reference/internals/metrics/) for additional detail.

#### Display runtime configuration in use by `influxd`

- Update `api/v2/config` endpoint to display the correct runtime configuration (for example, when you run `influxd print-config`).

> ### Bug fixes

Reviewer 4: Think we need this added: https://github.com/influxdata/influxdb/pull/23219
@kelseiv how are these usually populated? Is it manual?

Author: yes, @samhld, manual, unfortunately. We've had dreams of automating some of this process, some day.

- Ensure manual task runs can happen concurrently.
- Extend snapshot copy of backup to filesystems that do not support hard links.
- Detect misquoted tag values and return an error.
- Successfully handle errors returned by `Sketch.Merge`.
- Return `X-version` and `X-Build` headers for all requests to `/ping` endpoint.
- Add error when `meta.db` is missing.
- Sync TSI index file before closing.

> - Fix race condition that could cause `restore` command to fail.

Reviewer: Looking at the changelog:

`1. [de7f052](https://github.com/influxdata/influxdb/commit/de7f052): Chore: fix `dump_tsi` deadlock`

@DStrand1 this should have been marked 'fix' rather than 'chore'.

Docs team: text can be 'Fix potential deadlock in `influxd inspect dump-tsi`'

- To successfully restore a backup, use copy when a rename spans volumes.
- Disable use of jsonnet with `/api/v2/templates/apply`.
- Ensure that updating a check (`/checks/{checkID}`) does not require an owner ID.

> - Remove NATS for scraper processing. 

Reviewer: Worth mentioning that scrapers themselves still function fine?

Reviewer 4: Yes -- agreed!

```Remove NATS for scraper processing. Scrapers still work as they did before.```?

weight: 101
---

## v2.2 [2022-02-04]

### Features

#### Flux

> - Update to [Flux v0.150.0](/flux/v0.x/release-notes/#v?).

Author: check with Scott on Flux .150 release notes, we can link to later if need be

- (https://github.com/influxdata/influxdb/commit/b02c89e): Option to log flux queries cancelled because of server shutdown.

#### Miscellaneous upgrades

> https://github.com/influxdata/influxdb/commit/f4e9ae9): Build: upgrade protobuf library

Author: how many of the "build" updates below make sense to mention?

#### Task metadata

> (https://github.com/influxdata/influxdb/commit/504f0e4): Passing `type=basic` returns task metadata without query text

Author: prob need an example of this somewhere

(https://github.com/influxdata/influxdb/commit/cd0243d): Added replications queue management to launcher tasks
(https://github.com/influxdata/influxdb/commit/6b56af3): Mirror writes to registered replications
(https://github.com/influxdata/influxdb/commit/40d9587): Add replications queue scanner
(https://github.com/influxdata/influxdb/commit/ad52815): Add field for dropping data resulting in non-retryable errors to individual replications
(https://github.com/influxdata/influxdb/commit/9873ccd): Remote write function for replications
(https://github.com/influxdata/influxdb/commit/f05d013): Metrics collection for replications remote writes
(https://github.com/influxdata/influxdb/commit/3460f1c): Replication remote writes do not block local writes
(https://github.com/influxdata/influxdb/commit/28bcd41): Batch replications remote writes to avoid payload limit errors
(https://github.com/influxdata/influxdb/commit/a7a5233): Advance queue scanner periodically instead of every remote write

#### Performance?

(https://github.com/influxdata/influxdb/commit/fea3037): Configure perf tests with yaml

#### Metrics

> 

Author: how to access these metrics

(https://github.com/influxdata/influxdb/commit/dece95d): Tsm compaction metrics via prometheus
(https://github.com/influxdata/influxdb/commit/3a81166): Added metrics collection for replications

> (https://github.com/influxdata/influxdb/commit/6096ee2): Replications metrics include failure to enqueue

Author: move to replications?

(https://github.com/influxdata/influxdb/commit/0a74085): Point write requests have metrics
(https://github.com/influxdata/influxdb/commit/a74e051): Disk size metrics per shard
(https://github.com/influxdata/influxdb/commit/feb459c): Metrics for cache subsystem
(https://github.com/influxdata/influxdb/commit/edb21ab): Metrics for wal subsystem
(https://github.com/influxdata/influxdb/commit/b970e35): Remaining storage metrics from OSS engine

#### Miscellaneous updates

(https://github.com/influxdata/influxdb/commit/c51a0df): Error out when config file contains 1.x config values
(https://github.com/influxdata/influxdb/commit/afb167a): `query-memory-bytes` zero-value is unlimited

> (https://github.com/influxdata/influxdb/commit/f78f9ed): Api/v2/config endpoint displays runtime configuration

Author: add to API docs?

> (https://github.com/influxdata/influxdb/commit/4f74049): Add downgrade target for 2.1

Author: install doc

https://github.com/influxdata/influxdb/commit/2bace77): Unhandled errors returned by Sketch.Merge
https://github.com/influxdata/influxdb/commit/fa9ba8e): Duplicated X-version and X-Build headers for /ping endpoint
https://github.com/influxdata/influxdb/commit/8aa3a8f): Add causal error when meta.db is missing
https://github.com/influxdata/influxdb/commit/799d349): Sync index file before close
https://github.com/influxdata/influxdb/commit/5ce164f): Remove influx CLI output from CONTRIBUTING
https://github.com/influxdata/influxdb/commit/e4e1633): Replications remote writes do not block server shutdown
https://github.com/influxdata/influxdb/commit/39eeb3e): Fix race condition which could cause restore command to fail
https://github.com/influxdata/influxdb/commit/e5cbd27): Advance replications queue after successful remote writes
https://github.com/influxdata/influxdb/commit/4fd4bd0): Use copy when a rename spans volumes
https://github.com/influxdata/influxdb/commit/11c0081): Disable use of jsonnet with `/api/v2/templates/apply`
https://github.com/influxdata/influxdb/commit/0c30afd): Updating a check does not require an owner id

### Other

> 

Author: not sure how many of these others we want to mention, let's check with edge team

#### Technical preview: replicate data remotely

- Add the option to [Replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).

  This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature lets you create a durable subscription on a per bucket basis for the purposes of replicating data on-write from an InfluxDB OSS instance to InfluxDB Cloud.
  - Lets you store, analyze, and aggregate data locally while also forwarding newly arriving data to a centralized InfluxDB Cloud account.
  - If InfluxDB OSS loses connectivity to InfluxDB Cloud, the data gets sent when network connectivity is reestablished.
  - Configuration options include a maximum buffer size and data age restrictions to restrict the amount of data stored on disk.

#### Flux updates

- Update to [Flux v0.150.0](/flux/v0.x/release-notes/#v01500-2022-01-19).
- Add option to log Flux queries that are cancelled because of server shutdown.

#### Build maintenance

- Upgrade `protobuf` library.

#### Task metadata

- Add option to pass `type=basic` to [`/api/v2/tasks`](/influxdb/v2.1/api/#tag/Tasks) to return task metadata without the query text.

#### Metrics

> - Add Prometheus metrics for TSM compactions, including:

Reviewer 5: not sure that "Prometheus" is meaningful here.  They are Prometheus Exposition Format....   But the key point is that these are key metrics about the storage sub-system which are exposed now by InfluxDB....

Most of these were previously available with InfluxDB v1, but were not initially available with InfluxDB v2.  These are being re-introduced to assist with troubleshooting and other operational management.

Reviewer 4: Still at it!

weight: 101
---

## v2.2 [2022-03-29]

This release includes the following new [features](#features) and several [bug fixes](#bug-fixes).

### Features

- [Technical preview of remote data replication](#replicate-data-remotely-technical-preview)
- [Flux updates](#flux-updates)
- [Build maintenance](#build-maintenance)
- [Task metadata](#task-metadata)
- [Metrics](#metrics)
- [Display runtime configuration in use by `influxd`](#display-runtime-configuration-in-use-by-influxd)
- [Recover user credentials](#recover-user-credentials)
- [Security updates](#security-updates), including a new `hardening-enabled` option to enable additional security.

#### Technical preview: Durable subscription to replicate data

- Add the option to [Replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).

>   This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature allows you to create a durable subscription on a per bucket basis for the purposes of replicating data from an InfluxDB OSS instance to InfluxDB Cloud. 

Reviewer 4: I would just add that it's "replication on write". Like, "...replicating data on-write from an InfluxDB...."

>   This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature allows you to create a durable subscription on a per bucket basis for the purposes of replicating data from an InfluxDB OSS instance to InfluxDB Cloud. 

Reviewer 5: I think that URL should be /v2.2. no?

   - `storage_compaction`
   - `storage_compactions_active`
   - `storage_compactions_duration_seconds_bucket`
   - `storage_compactions_duration_seconds_count`
   - `storage_compactions_duration_seconds_sum`
   - `storage_compactions_queued`
   - `storage_compactions_total`

- Add the following metrics tracking when data fails to be added to the replication queue:
   - `PointsFailedToQueue`
   - `BytesFailedToQueue`

- Add the following metrics for writes, to indicate the status:
  - `pointsWriteRequested`
	- `pointsWriteOk`
	- `pointsWriteDropped`
	- `pointsWriteErr`
	- `timeout`

- Add disk size metrics per shard:
  - `total`: number of files per shard
  - `disk_bytes`: data size in bytes for each shard

- Add cache subsystem metrics:
  - `inuse_bytes`: Current memory consumption of cache
  - `diskBytes`: Size of most recent snapshot"
  - `latest_snapshot`: Unix time of most recent snapshot
  - `writes_total`: Counter of all writes to cache
  - `writes_err`: Counter of failed writes to cache
  - `writes_dropped`: Counter of writes (with any dropped points) to cache

- Add WAL subsystem metrics:
  - `size`: Size of WAL in bytes
  - `writes`: Number of write attempts to the WAL

>   - `writes_err`: Number of failed write attempts to the WAL

Reviewer 6: Suggest removing the extra vertical space between these metrics kinds.

See [InfluxDB OSS metrics](/influxdb/v2.1/reference/internals/metrics/) for additional detail.

#### Display runtime configuration in use by `influxd`

> - Update `api/v2/config` endpoint to display the correct runtime configuration (for example, when you run `influxd print-config`).

Reviewer 4: Change "Update" to "Added"

> - Update `api/v2/config` endpoint to display the correct runtime configuration (for example, when you run `influxd print-config`).

Reviewer 4: ```suggestion
- Added `api/v2/config` endpoint to display the runtime configuration (for example, when you run `influxd print-config`). This endpoint lets you review runtime configuration while the instance is live.
```

#### Recover user credentials

To recover user credentials, use [`influx recovery user list`](/influxdb/v2.2/reference/cli/influxd/recovery/user/list/) to retrieve a list of users, and then use [`influx recovery user update`](/influxdb/v2.2/reference/cli/influxd/recovery/user/update/) to update the password for a specified user.

### Security updates

> - Add the [`hardening-enabled`](/influxdb/v2.2/security/enable-hardening.md) option to limit flux/pkger HTTP requests. The new `hardening-enabled` option ensures that InfluxDB first verifies the IP address of the URL is not a private.

Reviewer 6: ```suggestion
- Add the [`hardening-enabled`](/influxdb/v2.2/security/enable-hardening.md) option to limit flux/pkger HTTP requests. The new `hardening-enabled` option ensures that InfluxDB first verifies the IP address of the URL is not private.
```

  By default, Flux HTTP and template fetching requests are allowed to access localhost and private IP addresses.
  - Disable use of jsonnet with `/api/v2/templates/apply`.
This prevents crafted authenticated requests from exfiltrating files accessible to the user InfluxDB runs as.
- Add read permissions check for querying data.

> This prevents authenticated requests using a write-only token from reading data

Reviewer 6: ```suggestion
  This prevents authenticated requests using a write-only token from reading data
```

  - Make substring check bounds correctly.
    This prevents authenticated queries from crashing the Flux engine.
- The cumulative security fixes for [Go 1.17.8](https://go.dev/doc/devel/release#go1.17.minor) since Go 1.17.2 are included in this release.
  This addresses an issue in the InfluxDB test suite.

### Bug fixes

- Ensure manual task runs can happen concurrently.
- Extend snapshot copy of backup to filesystems that do not support hard links.
- Detect misquoted tag values and return an error.
- Fix potential deadlock in `influxd inspect dump-tsi`.
- Successfully handle errors returned by `Sketch.Merge`.
- Return `X-version` and `X-Build` headers for all requests to `/ping` endpoint.
- Add error when `meta.db` is missing.
- Sync TSI index file before closing.
- Fix race condition that could cause `restore` command to fail.
- To successfully restore a backup, use copy when a rename spans volumes.
- Disable use of jsonnet with `/api/v2/templates/apply`.
- Ensure that updating a check (`/checks/{checkID}`) does not require an owner ID.
- Remove NATS for scraper processing. Note, this fix does not alter scraper functionality--scrapers still work as they did before.

>   - ``nats-port`` and ``nats-max-payload-bytes`` flags have been deprecated.

Reviewer 6: ```suggestion
  - `nats-port` and `nats-max-payload-bytes` flags have been deprecated.
```

  -  NATS is no longer embedded in InfluxDB. Because InfluxDB no longer requires a port for NATS, port conflict issues are reduced.
- Resolve the issue that prevented the browser from tracking the cookie `expiry` correctly, causing the cookie to expire automatically when restarting the browser or changing tabs. Now, the cookie is correctly preserved.

> - Allow unlimited Flux http calls. Previously, http requests failed silently after 100MB of data transfer.

Reviewer 6: ```suggestion
- Allow unlimited Flux HTTP calls. Previously, HTTP requests failed silently after 100MB of data transfer.
```

weight: 101
---

## v2.2 [TBD]

### Features

#### Replicate data remotely

> - Add the option to [replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/cloud/write-data/replication.md).

Author: ```suggestion
- Add the option to [replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/cloud/write-data/replication).
```

> - Add the option to [replicate data from InfluxDB OSS to InfluxDB Cloud](content/influxdb/cloud/write-data/replication.md).

Author: ```suggestion
- Add the option to [replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/cloud/write-data/replication.md).
```

This feature helps you to collect data locally from multiple InfluxDB OSS edge deployments, analyze and aggregate the data, and then send the data to one or more buckets on a remote InfluxDB Cloud instance. If InfluxDB OSS loses connectivity to InfluxDB Cloud, the data is sent when network connectivity is restored.

#### Flux updates

- Update to [Flux v0.150.0](/flux/v0.x/release-notes/#v01500-2022-01-19).

> - Option to log Flux queries that are cancelled because of server shutdown.

Author: Is this option to log Flux queries something that folks must turn on somewhere, or does this happen automatically? how can log be accessed?

#### Build maintenance

- Upgrade `protobuf` library.

> - https://github.com/influxdata/influxdb/commit/5a1e375): Build: allow to build on FreeBSD

Author: does this mean we allow installs of 2.2 on FreeBSD? do we need to update install instructions for this?

Reviewer 5: I think this just enables the build process.  We do not plan to support FreeBSD.  So, I think we can skip the docs for this.

Reviewer 2: Removed:

- Build: allow to build on FreeBSD

#### Task metadata

> - (https://github.com/influxdata/influxdb/commit/504f0e4): Passing `type=basic` returns task metadata without query text.

Author: Is this an option the user can select or pass somewhere, or solely behind the scenes?

Reviewer 3: @kelseiv It's in the public API (https://github.com/influxdata/openapi/blob/6d3d0c2267e5fd55d6b832d491e2a14ba4cf2c96/contracts/ref/oss.yml#L11491), though the primary user seems to be the InfluxDB UI.

#### Metrics

> (https://github.com/influxdata/influxdb/commit/3a81166): Added metrics collection for replications

Author: @pierwill to be sure, did we add these replication metrics for troubleshooting into a task?: https://github.com/influxdata/docs-v2/issues/3459#issuecomment-988983567

Reviewer 5: I think this is a little confusing.  This indicates that new metrics are now exposed which describe the details of the Durable Subscription -- how much data is being replicated, how much is enqueued, etc.  What this means is that if you use a scraper to gather metrics from InfluxDB OSS....there will be a new set of metrics related to this feature.  If you use Telegraf to gather metrics from InfluxDB OSS...there will be a new set of metrics related to this feature.

Reviewer 6: > @pierwill to be sure, did we add these replication metrics for troubleshooting into a task?: [#3459 (comment)](https://github.com/influxdata/docs-v2/issues/3459#issuecomment-988983567)

Created a breakout issue: https://github.com/influxdata/docs-v2/issues/3769

> (https://github.com/influxdata/influxdb/commit/dece95d): TSM compaction metrics via prometheus

Author: should these Prometheus metrics be added [here](https://docs.influxdata.com/influxdb/v2.1/reference/prometheus-metrics/#resulting-line-protoco)?

Reviewer 2: @kelseiv create this issue addressing Prometheus metrics:  https://github.com/influxdata/docs-v2/issues/3782

> (https://github.com/influxdata/influxdb/commit/dece95d): TSM compaction metrics via prometheus

Author: ```suggestion
- Add Prometheus metrics for TSM compactions, including:
   - `storage_compaction`
   - `storage_compactions_active`
   - `storage_compactions_duration_seconds_bucket`
   - `storage_compactions_duration_seconds_count`
   - `storage_compactions_duration_seconds_sum`
   - `storage_compactions_queued`
   - `storage_compactions_total`
```

- Add the following metrics when data fails to be added to the replication queue:
  `PointsFailedToQueue`
  `BytesFailedToQueue`

- Add the following metrics for writes, to indicate the status:
	- `pointsWriteRequested`
	- `pointsWriteOk`
	- `pointsWriteDropped`
	- `pointsWriteErr`

> 	- `timeout`

Author: @jstirnaman lines 40-49 are the metrics we chatted about this morning that you guys thought may be covered by the rejected points content...to be sure, anything to clarify or add in context?

Would it make sense to send folks here?:

https://docs.influxdata.com/influxdb/v2.1/write-data/troubleshoot/#troubleshoot-rejected-points
Or here:
https://docs.influxdata.com/influxdb/v2.1/api/#tag/Write

Reviewer 3: Metrics doc will be at https://docs.influxdata.com/influxdb/v2.1/reference/internals/metrics (I hadn't merged it yet). We'll need to add these new metrics there. I suppose we should also mention them in https://docs.influxdata.com/influxdb/v2.1/write-data/troubleshoot/#troubleshoot-rejected-points like you suggested.

- Add disk size metrics per shard:
  - `total`: number of files per shard
  - `disk_bytes`: data size in bytes for each shard

- Add cache subsystem metrics:
  - `inuse_bytes`: Current memory consumption of cache
  - `diskBytes`: Size of most recent snapshot"
  - `latest_snapshot`: Unix time of most recent snapshot.
  - `writes_total`: Counter of all writes to cache
  - `writes_err`: Counter of failed writes to cache.
  - `writes_dropped`: Counter of writes (with any dropped points) to cache.

- Add WAL subsystem metrics:
  - `size`: Gauge of size of WAL in bytes
  - `writes`: Number of write attempts to the WAL
  - `writes_err`: Number of failed write attempts to the WAL

> 

Author: does it make sense to add these storage metrics [here](http://localhost:1313/influxdb/v2.1/reference/internals/storage-engine/)?
Or are these only a subset of all storage metrics?

#### Display runtime configuration in use by `influxd`

> - Update `api/v2/config` endpoint to display the correct runtime configuration (for example, when you run `influxd print-config`).

Author: I can't find the `api/v2/config` endpoint in the docs--is this name accurate/am I missing?

Reviewer 3: @kelseiv It's new, as of a couple of months ago; won't be in v2.1 spec.

weight: 101
---

## v2.2 [TBD]

This release includes the following new [features](#features) and several [bug fixes](#bug-fixes).

### Features

- [Technical preview of replicating data remotely](#replicate-data-remotely-technical-preview)
- [Flux updates](#flux-updates)
- [Build maintenance](#build-maintenance)
- [Task metadata](#task-metadata)
- [Metrics](#metrics)
- [Display runtime configuration in use by `influxd`](#display-runtime-configuration-in-use-by-influxd)

#### Replicate data remotely (technical preview)

- Add the option to [replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).

>   In this **technical preview**, you can replicate inbound data using the HTTP API. Support for replicating data via tasks (for example, downsampling) is targeted prior to the official release of this feature.

Reviewer 3: Do we have a definition for "technical preview"?

> ...is targeted prior to the official release

What does "is targeted prior" mean here?

Author: Good question, @jstirnaman - verifying with @timhallinflux on the official "technical preview" definition. 
Believe the intention is to give folks half of a pie to start eating (replicate inbound data w API), but know they’re going to ask where the other half is (replicate processed data outside of API too), so we’re getting ahead of that…

agree the "is targeted prior" isn't super explicit...for purposes of the release notes, we should prob omit[ future](https://developers.google.com/style/future context) from official docs, and leave that part in release announcements, blogs, marketing teasers, and those channels

Reviewer 2: Removed:

Support for replicating data via tasks (for example, downsampling) is targeted prior to the official release of this feature.

>   This feature helps you to collect data locally from multiple InfluxDB OSS edge deployments, analyze and aggregate the data, and then send the data to one or more buckets on a remote InfluxDB Cloud instance. If InfluxDB OSS loses connectivity to InfluxDB Cloud, the data is sent when network connectivity is restored.

Reviewer 5: I think this is a little bit confusing. We used to have a subscription mechanism in v1.  The difference is that the the subscription wasn't durable.  Meaning, it was fire and forget.  If the upstream connectivity was lost, the data was not stored, buffered or preserved in any way.   What we are introducing here is more of a "durable subscription."  I would propose:

Technical  Preview: Durable Subscription

The feature helps allows you to create a durable subscription on a per bucket basis for the purposes of replicating data from  InfluxDB OSS instance to InfluxDB Cloud.  This allows users to store, analyze and aggregate data locally while also forwarding newly arriving data to a centralized InfluxDB Cloud account.  If InfluxDB OSS loses connectivity to InfluxDB Cloud, the data is sent when network connectivity is restored.  Configuration options include a maximum buffer size and data age restrictions to restrict the amount of data stored on disk.

Limitations:
Currently only data received directly via the v2 write API is stored locally within the InfluxDB OSS instance and replicated.  This means that data created using Flux tasks is not replicated.  This limitation is intended to be resolved as part transitioning this feature from technical preview to production ready.  Additional performance enhancements are also planned.

If you add a durable subscription to an existing bucket, any data already present within the bucket is not replicated.

Reviewer 5: Feedback:  
We are looking for feedback regarding this technical preview feature.  You can contact us via slack (link to community slack) or our community site (community.influxdata.com).

Reviewer 2: added this issue: https://github.com/influxdata/docs-v2/issues/3777 to add definition of technical preview

Reviewer 2: Borrowed Tim's wording to clarify what this durable subscription feature is.

#### Flux updates

- Update to [Flux v0.150.0](/flux/v0.x/release-notes/#v01500-2022-01-19).
- Option to log Flux queries that are cancelled because of server shutdown.

#### Build maintenance

- Upgrade `protobuf` library.
- https://github.com/influxdata/influxdb/commit/5a1e375): Build: allow to build on FreeBSD

#### Task metadata

- Add option to pass `type=basic` to [`/api/v2/tasks`](/influxdb/v2.1/api/#tag/Tasks) to return task metadata without the query text.

#### Metrics

(https://github.com/influxdata/influxdb/commit/3a81166): Added metrics collection for replications
- Add Prometheus metrics for TSM compactions, including:
   - `storage_compaction`
   - `storage_compactions_active`
   - `storage_compactions_duration_seconds_bucket`
   - `storage_compactions_duration_seconds_count`
   - `storage_compactions_duration_seconds_sum`
   - `storage_compactions_queued`
   - `storage_compactions_total`

> - Add the following metrics when data fails to be added to the replication queue:

Reviewer 6: ```suggestion
- Add the following metrics tracking when data fails to be added to the replication queue:
```

  `PointsFailedToQueue`
  `BytesFailedToQueue`

- Add the following metrics for writes, to indicate the status:
	- `pointsWriteRequested`
	- `pointsWriteOk`
	- `pointsWriteDropped`
	- `pointsWriteErr`
	- `timeout`

- Add disk size metrics per shard:
  - `total`: number of files per shard
  - `disk_bytes`: data size in bytes for each shard

- Add cache subsystem metrics:
  - `inuse_bytes`: Current memory consumption of cache
  - `diskBytes`: Size of most recent snapshot"

>   - `latest_snapshot`: Unix time of most recent snapshot.

Reviewer 6: Let's make end punctuation consistent here.

  - `writes_total`: Counter of all writes to cache
  - `writes_err`: Counter of failed writes to cache.
  - `writes_dropped`: Counter of writes (with any dropped points) to cache.

- Add WAL subsystem metrics:

>   - `size`: Gauge of size of WAL in bytes

Reviewer 6: ```suggestion
  - `size`: Size of WAL in bytes
```

  - `writes`: Number of write attempts to the WAL
  - `writes_err`: Number of failed write attempts to the WAL

#### Display runtime configuration in use by `influxd`

- Update `api/v2/config` endpoint to display the correct runtime configuration (for example, when you run `influxd print-config`).

### Bug fixes

> - Update to ensure manual task runs can happen concurrently.

Reviewer 6: ```suggestion
- Ensure manual task runs can happen concurrently.
```

- Extend snapshot copy of backup to filesystems that do not support hard links.
- Detect misquoted tag values and return an error.
- Successfully handle errors returned by `Sketch.Merge`.
- Return `X-version` and `X-Build` headers for all requests to `/ping` endpoint.
- Add error when `meta.db` is missing.
- Sync TSI index file before closing.
- Fix race condition that could cause `restore` command to fail.

> - To successfully restore a backup, use copy when a rename spans volumes.

Reviewer 6: Not sure what the update is here...

weight: 101
---

## v2.2 [TBD]

This release includes the following new [features](#features) and several [bug fixes](#bug-fixes).

### Features

- [Technical preview of replicating data remotely](#replicate-data-remotely-technical-preview)
- [Flux updates](#flux-updates)
- [Build maintenance](#build-maintenance)
- [Task metadata](#task-metadata)
- [Metrics](#metrics)
- [Display runtime configuration in use by `influxd`](#display-runtime-configuration-in-use-by-influxd)

> #### Replicate data remotely (technical preview)

Author: ```suggestion
#### Technical preview: Durable subscription to replicate data
```

> - Add the option to [Replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).

Author: ```suggestion
```

>   This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature allows you to create a durable subscription on a per bucket basis for the purposes of replicating data from an InfluxDB OSS instance to InfluxDB Cloud. This allows users to store, analyze and aggregate data locally while also forwarding newly arriving data to a centralized InfluxDB Cloud account. If InfluxDB OSS loses connectivity to InfluxDB Cloud, the data is sent when network connectivity is restored. Configuration options include a maximum buffer size and data age restrictions to restrict the amount of data stored on disk.

Author: ```suggestion
This [**technical preview**](/influxdb/v2.1/reference/glossary/#technical-preview) feature lets you create a durable subscription on a per bucket basis, to replicate data from an InfluxDB OSS instance to InfluxDB Cloud. Provides a way to store, analyze and aggregate data locally, while forwarding newly arriving data to a centralized InfluxDB Cloud account. If InfluxDB OSS loses connectivity to InfluxDB Cloud, local data is retained and sent to Cloud when network connectivity is restored. To restrict the amount of data stored on disk, configure maximum buffer size and data age restrictions.
  
- Add the option to [Replicate data from InfluxDB OSS to InfluxDB Cloud](/influxdb/v2.2/write-data/replication).
```

Author: let's specify the exact config settings here
